core/modules/admin_bar/admin_bar.inc

```diff
@@ -826,6 +836,10 @@ function admin_bar_theme_settings() {
 function admin_bar_theme_settings_submit($form, &$form_state) {
   $components = &$form_state['values']['components'];
 
+  // Disable back to site if icon bar is also disabled.
+  if (!$components['admin_bar.icon']) {
+    $components['admin_bar.back_to_site_link'] = FALSE;
```

Comment:
So, here's an idea about the behavior of this checkbox. Typically, when a checkbox depends on another and we hide it, we don't want to change the value of the hidden checkbox. So when/if the user changes their mind, they can re-enable to top checkbox and see that the lower one has the same value as before. So, my suggestion is to leave this value alone as is, and modify the preprocess function below to check for both, the `admin_bar.back_to_site_link` component AND the `admin_bar.icon` one. See my next  comment below.

If this sounds good to you, then these 4 lines can be removed.

Reply:
I copied this from the lines underneath it which already has the same behavior.   The search bar is disabled when the management menu is deselected and setting saved.

Reply:
You are right. Then let's just keep it as is.